### DISCUSSION TITLE
Samba/SMB connected shares as music library ...

### DISCUSSION THREAD

**Discussion by buforded** [avantrec/soco-cli#70] [Q&A]
Hello,

I pored over the documentation and did not see any mention of SMB/Samba but I do see discussion of the "music library" so I am unsure how soco-cli is handling music files that are in a music library connected over the network (SMB/Samba).

To be specific, in the official Sonos desktop app:

Music Library -> Folders -> //192.168.1.10/musicshare

Is it currently possible to access these tracks ?

I will note, as we discuss this, that my own files are all WAV files so there is no metadata - no "track names" or "album names" or "artist" - it's all just filename and nothing else.  Sonos desktop app has always handled directories and let me choose not just a file, but an entire directory to add to the queue.  But this is very different than adding an "album" or whatever ... 

Thanks very much.

**Comment by pwt** [comment]
That's a good question: I don't know if there's a way in SoCo to search using just file/folder names, without metadata. I'll look into it when I get a chance.

**Comment by buforded** [comment]
Thank you.

To be clear - my major concern is, can I access a SMB/samba library through soco-cli ?  It's not clear one way or another from the documentation ...

**Reply by pwt** [reply]
I think the SoCo-CLI documentation is clear: it uses the term Music Library in exactly the way Sonos defines it.

A Sonos Music Library is a collection of media typically served to Sonos by one or more SMB shares. These shares are added to Sonos using one of the Sonos apps, and Sonos searches the shares and indexes their contents.

The SoCo-CLI Music Library functions allow these indexes inside Sonos to be searched, and content from them added to the Sonos queue. It's then up to Sonos to play the media by obtaining it directly via the shares.

SoCo-CLI doesn't itself connect to any SMB shares -- it's all done via Sonos.

Does that help?
